server/lib/notifications/notification-api.ts

```diff
@@ -25,8 +27,14 @@ export class NotificationApi extends HttpApi {
 }
 
 async function clearNotifications(ctx: RouterContext) {
-  // FIXME: this needs to use the service and needs to notify all connected clients
-  await clear(ctx.session!.userId)
+  const { body } = validateRequest(ctx, {
+    body: Joi.object<ClearNotificationsServerBody>({
+      timestamp: Joi.number().required(),
```

Comment:
Timestamp should be optional because the client may not have any server notifications to pull a timestamp from. In the case that it isn't provided, the server should use its current time.

Since the client won't necessarily know what timestamp was used, the server should also return it in the response body (so this will be a `200` instead of a `204`).

Reply:
Is it really necessary to include it in the response body of the HTTP request if we're already including it in the websocket event that is published to all connected clients after the notifications are cleared?

Reply:
Not passing it back assumes a sort of implementation on the server that may not always be true, in general I'd prefer to think about these APIs in isolation from any other behaviors that may occur. It's also difficult to map a particular request to a particular websocket broadcast, those things are not connected like a request/response are.